src/app/pages/office/cat-a-mod2/office.cat-a-mod2.page.ts

```diff
@@ -1,35 +1,287 @@
+import {
+  AddSafetyAndBalanceComment,
+} from '@store/tests/test-data/cat-a-mod2/safety-and-balance/safety-and-balance.cat-a-mod2.actions';
+import { EyesightTestAddComment } from '@store/tests/test-data/common/eyesight-test/eyesight-test.actions';
+import { OutcomeBehaviourMapProvider } from '@providers/outcome-behaviour-map/outcome-behaviour-map';
+import { SetActivityCode } from '@store/tests/activity-code/activity-code.actions';
+import { FaultSummaryProvider } from '@providers/fault-summary/fault-summary';
+import { FaultCountProvider } from '@providers/fault-count/fault-count';
+import {
+  safetyAndBalanceQuestionsExist,
+} from '@store/tests/test-data/cat-a-mod2/safety-and-balance/safety-and-balance.cat-a-mod2.selector';
+import {
+  getSafetyAndBalanceQuestions,
+} from '@store/tests/test-data/cat-a-mod2/test-data.cat-a-mod2.selector';
+import { getNewTestStartTime } from '@shared/helpers/test-start-time';
+import { SetStartDate }
+  from '@store/tests/journal-data/common/test-slot-attributes/test-slot-attributes.actions';
+import { CommentSource, FaultSummary } from '@shared/models/fault-marking.model';
+import { ActivityCodeModel, activityCodeModelList } from '@shared/constants/activity-code/activity-code.constants';
+import {
+  CommonOfficePageState,
+  OfficeBasePageComponent,
+} from '@shared/classes/test-flow-base-pages/office/office-base-page';
 import { Router } from '@angular/router';
-import { RouteByCategoryProvider } from '@providers/route-by-category/route-by-category';
-import { JOURNAL_PAGE, TestFlowPageNames } from '../../page-names.constants';
+import {
+  AdditionalInformationChanged,
+  CandidateDescriptionChanged,
+  IdentificationUsedChanged,
+  IndependentDrivingTypeChanged,
+  RouteNumberChanged,
+  WeatherConditionsChanged,
+} from '@store/tests/test-summary/test-summary.actions';
+import { behaviourMap } from '../office-behaviour-map.cat-a-mod2';
+import {
+  OfficeViewDidEnter,
+  TestStartDateChanged,
+} from '../office.actions';
+
+interface CatAMod2OfficePageState {
+  testCategory$: Observable<TestCategory>;
+  displayModeOfTransport$: Observable<boolean>;
+  modeOfTransport$: Observable<ModeOfTransport>;
+  displayDrivingFaultComments$: Observable<boolean>;
+  safetyAndBalanceQuestions$: Observable<SafetyQuestionResult[]>;
+  displaySafetyAndBalance$: Observable<boolean>;
+}
+
+type OfficePageState = CommonOfficePageState & CatAMod2OfficePageState;
 
 @Component({
-  selector: 'app-office-cat-a-mod2',
-  templateUrl: './office.cat-a-mod2.page.html',
-  styleUrls: ['./office.cat-a-mod2.page.scss'],
+  selector: '.office-cat-a-mod2-page',
+  templateUrl: 'office.cat-a-mod2.page.html',
+  styleUrls: ['../office.page.scss'],
 })
-export class OfficeCatAMod2Page implements OnInit {
+export class OfficeCatAMod2Page extends OfficeBasePageComponent {
+  pageState: OfficePageState;
+  pageSubscription: Subscription;
+  form: FormGroup;
```

Comment:
This can be removed

Reply:
I thought we needed the pageState as it consists of a few fields bespoke to mod2 spread with the commonOfficePageState?

Reply:
I left the comment on the ```form: FormGroup;``` line

Reply:
ah sorry thought it was for lines 97-99

Reply:
This has not been resolved